### PR TITLE
change CommandLineApplication.Name for pack and restore to include 'dotnet'

### PR DESCRIPTION
**Pull request by jonsequitur**
This addresses a few inconsistencies in the "Usage" line of the help output for the `pack` and `restore` commands relative to most of the commands.

It addresses #4813.

### CODE REVIEW

File: test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs [hunk 1]
```diff
@@ -65,5 +66,33 @@ public void WhenHelpOptionIsPassedToDotnetItPrintsUsage(string helpArg)
...
+        [InlineData("run", "Usage: dotnet run [options] [[--] <additional arguments>...]]")]
+        [InlineData("test", "Usage: dotnet test [arguments] [options] [args]")]
+        [InlineData("migrate", "Usage: dotnet migrate [arguments] [options]")]
+        [InlineData("clean", "Usage: dotnet clean [arguments] [options] [args]")]
+        [InlineData("sln", "Usage: dotnet sln [arguments] [options] [command]")]
+        [InlineData("sln add", "Usage: dotnet sln <SLN_FILE> add [options] [args]")]
+        [InlineData("sln list", "Usage: dotnet sln <SLN_FILE> list [options]")]
+        [InlineData("sln remove", "Usage: dotnet sln <SLN_FILE> remove [options] [args]")]
+        [InlineData("nuget", "Usage: dotnet nuget [options] [command]")]
+        // [InlineData("msbuild", "")]
```
Comment (jonsequitur) [inline test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs]: #4813 mentions that there's also an issue with `dotnet msbuild -h`. The output there is quite inconsistent with the others. I included the commented out code as a placeholder pending further discussion. Is this something we want to try to address now?

```
$ dotnet msbuild -h
Microsoft (R) Build Engine version 15.1.545.13942
Copyright (C) Microsoft Corporation. All rights reserved.

Syntax:              MSBuild.exe [options] [project file | directory]

Description:         Builds the specified targets in the project file. If
                     a project file is not specified, MSBuild searches the
                     current working directory for a file that has a file
                     extension that ends in "proj" and uses that file.  If
                     a directory is specified, MSBuild searches that
                     directory for a project file.

...
```

File: test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs [hunk 2]
```diff
@@ -65,5 +66,33 @@ public void WhenHelpOptionIsPassedToDotnetItPrintsUsage(string helpArg)
...
+        // [InlineData("vstest", "")]
```
Comment (jonsequitur) [inline test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs]: Comment included as a placeholder for further discussion.

The output here is also inconsistent with the others. 

It doesn't understand the `-h` option: 

```
$ dotnet vstest -h
Microsoft (R) Test Execution Command Line Tool Version 15.0.0.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Error: The test source file "/Users/jonsequitur/dev/cli/-h" provided was not found.
```

The `--help` option is closer to the others but still inconsistent:

```
$ dotnet vstest --help
Microsoft (R) Test Execution Command Line Tool Version 15.0.0.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Usage: vstest.console.exe [Arguments] [Options] [[--] <args>...]]

Description: Runs tests from the specified files.
```

**Comment by jonsequitur** [comment]
@piotrpMSFT @livarcocc @blackdwarf

File: test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs [hunk 3]
```diff
@@ -65,5 +66,35 @@ public void WhenHelpOptionIsPassedToDotnetItPrintsUsage(string helpArg)
...
+        [InlineData("publish", "Usage: dotnet publish [arguments] [options] [args]")]
+        [InlineData("run", "Usage: dotnet run [options] [[--] <additional arguments>...]]")]
+        [InlineData("test", "Usage: dotnet test [arguments] [options] [args]")]
+        [InlineData("migrate", "Usage: dotnet migrate [arguments] [options]")]
+        [InlineData("clean", "Usage: dotnet clean [arguments] [options] [args]")]
+        [InlineData("sln", "Usage: dotnet sln [arguments] [options] [command]")]
+        [InlineData("sln add", "Usage: dotnet sln <SLN_FILE> add [options] [args]")]
+        [InlineData("sln list", "Usage: dotnet sln <SLN_FILE> list [options]")]
+        [InlineData("sln remove", "Usage: dotnet sln <SLN_FILE> remove [options] [args]")]
+        [InlineData("nuget", "Usage: dotnet nuget [options] [command]")]
```
Comment (eerhardt) [inline test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs]: Are we concerned with new commands being added that don't know to go and update this?

ex. `dotnet cache`.  How do we ensure new commands are consistent?

Comment (jonsequitur) [inline test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs]: I don't think that with the current implementation of `CommandLineApplication`, there's a clean way to ensure this. This is a common theme in the command line parsing bugs and behavioral inconsistencies I've seen. Each command under `dotnet` has too much control. This is also why we have no true unit tests over our command line parsing code. I tried starting with a unit test in this case but the behavior in question was all specified at the individual commands so in the interest of keeping the change small, I used an integration test.

The command line parsing work I've been doing for v2 is meant to address these things.

File: test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs (hunk 1 above)
Comment (jonsequitur) [inline test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs]: @blackdwarf Any thoughts on this?

File: test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs (hunk 2 above)
Comment (jonsequitur) [inline test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs]: @blackdwarf Any thoughts on this?

File: test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs (hunk 1 above)
Comment (blackdwarf) [inline test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs]: @jonsequitur if we are talking about the inconsistency in how help is shown (not the switch for it), it is by design. The idea behind `dotnet msbuild` is that it gives you access to MSBuild in all its...well...glory. If we see people getting confused by this, we can investigate fixing it in some way.

File: test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs (hunk 2 above)
Comment (blackdwarf) [inline test/dotnet-help.Tests/GivenThatIWantToShowHelpForDotnetCommand.cs]: @jonsequitur same as above. That is why we separated these commands into a different bucket in the general help screen.

**Comment by jonsequitur** [comment]
The new parser work supersedes this PR so there's no need to port it to master.

**Pull request closed without merging**